File: app/src/main/java/com/example/smoiapp001/database/models/TransactionEntry.java
package com.example.smoiapp001.database.models;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import com.example.smoiapp001.utilities.DateConverter;

import java.util.Date;
import java.util.HashMap;
import java.util.Map;

@Entity(tableName = "transaction")
public class TransactionEntry {

    @PrimaryKey(autoGenerate = true)
    private int    id;
    private String description;
    private float  cost;
    private Date   date;

    public TransactionEntry(String description, float cost, Date date) {
        this.description = description;
        this.cost        = cost;
        this.date        = date;
    }

    public Map<String, Object> toMap() {
        HashMap<String, Object> result = new HashMap<>();
        result.put("id", id);
        result.put("description", description);
        result.put("cost", cost);
        result.put("date", DateConverter.toTimestamp(date));

        return result;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public float getCost() {
        return cost;
    }

    public void setCost(float cost) {
        this.cost = cost;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }
}

File: app/src/main/java/com/example/smoiapp001/database/models/TransactionEntry.kt
package com.example.smoiapp001.database.models

import androidx.room.Entity
import androidx.room.PrimaryKey

import com.example.smoiapp001.utilities.DateUtils

import java.util.Date
import java.util.HashMap

@Entity(tableName = "transaction")
class TransactionEntry(var description: String?, var cost: Float, var date: Date) {

    @PrimaryKey(autoGenerate = true)
    var id: Int = 0

    fun toMap(): Map<String, Any> {
        val result = HashMap<String, Any>()
        result["id"] = id
        result["description"] = description!!
        result["cost"] = cost
        result["date"] = DateUtils.toTimestamp(date)

        return result
    }
}
